edb/server/compiler/compiler.py

```diff
@@ -1822,17 +1836,133 @@ def _compile_ql_query(
 
     result_cardinality = enums.cardinality_from_ir_value(ir.cardinality)
 
+    # This low-hanging-fruit is temporary; persistent cache should cover all
+    # cacheable cases properly in future changes.
+    use_persistent_cache = (
+        ctx.persistent_cache and
+        cacheable and
+        not ctx.bootstrap_mode and
+        script_info is None
+    )
+
     sql_res = pg_compiler.compile_ir_to_sql_tree(
         ir,
         expected_cardinality_one=ctx.expected_cardinality_one,
         output_format=_convert_format(ctx.output_format),
         backend_runtime_params=ctx.backend_runtime_params,
         expand_inhviews=options.expand_inhviews,
+        detach_params=bool(use_persistent_cache and debug.flags.func_cache),
```

Comment:
Shouldn't need the `bool`, right?

Reply:
Right if we can let mypy think `Flag` is a `bool`:

```
edb/server/compiler/compiler.py:1854:23: error: Argument "detach_params" to "compile_ir_to_sql_tree" has incompatible type "Flag | bool"; expected "bool"  [arg-type]
```